src/store.c

```diff
@@ -256,11 +258,19 @@ int log_sample(DIR *proc,
                         char t[32];
                         struct ps_struct *parent;
 
-                        ps->next_ps = ps->next_running = new0(struct ps_struct, 1);
-                        if (!ps->next_ps)
+                        /* find the insertion point for the last item */
+                        struct ps_struct **ps_next = &ps->next_ps;
+                        while (*ps_next) {
+                                ps_next = &(*ps_next)->next_ps;
+                        }
+
+                        assert(!*ps_next);
```

Comment:
I think this assert is a bit paranoid. The while above ensures *ps_next is false by the time we get here.